newIDE/app/src/SceneEditor/index.js

```diff
@@ -979,6 +980,45 @@ export default class SceneEditor extends React.Component<Props, State> {
     done(true);
   };
 
+  canObjectOrGroupBeGlobal = (
+    i18n: I18nType,
+    objectOrGroupName: string
+  ): boolean => {
+    const { layout, project } = this.props;
+    const layoutName = layout.getName();
+    const layoutsWithObjectOrGroupWithSameName: Array<string> = mapFor(
+      0,
+      project.getLayoutsCount(),
+      i => {
+        const otherLayout = project.getLayoutAt(i);
+        const otherLayoutName = otherLayout.getName();
+        if (layoutName !== otherLayoutName) {
+          if (otherLayout.hasObjectNamed(objectOrGroupName)) {
+            return otherLayoutName;
+          }
+          const groupContainer = otherLayout.getObjectGroups();
+          if (groupContainer.has(objectOrGroupName)) {
+            return otherLayoutName;
+          }
+        }
+        return null;
+      }
+    ).filter(Boolean);
+
+    if (layoutsWithObjectOrGroupWithSameName.length > 0) {
+      showWarningBox(
+        i18n._(
+          t`Making "${objectOrGroupName}" global would conflict with the following scenes that have a group or an object with the same name:${'\n\n - ' +
```

Comment:
should we add a line "delete them first if you really want to set this object global"?

Reply:
I think it would make more sense to ask the user to rename them.  We don't want the user to delete their objects and (if they hit yes twice) their event logic.   Then once the global objects are added, the user can do a find/replace to change the event logic to use the global object.